creator-node/src/services/stateMachineManager/makeOnCompleteCallback.ts

```diff
@@ -131,13 +132,23 @@ const enqueueJobs = async (
   queueToAddTo: Queue,
   queueNameToAddTo: TQUEUE_NAMES,
   triggeredByQueueName: TQUEUE_NAMES,
+  maxWaitingJobs: number,
   triggeredByJobId: string,
   logger: Logger
 ) => {
   logger.info(
     `Attempting to add ${jobs?.length} jobs in bulk to queue ${queueNameToAddTo}`
   )
 
+  // Don't add to the queue if the queue is already backed up (i.e., it has too many waiting jobs)
+  const numWaitingJobs = await queueToAddTo.getWaitingCount()
```

Comment:
any possibility of queue not existing or this function throwing?

Reply:
don't think so -- if it didn't exist then it would throw later when we try to call `addBulk` on it, which hasn't ever happened (we would've seen logs for the catch statement here -- `"Failed to bulk-add jobs"` -- which I just double-checked are non-existent on loggly)

Reply:
coo